report/src/index.ts

```diff
@@ -0,0 +1,242 @@
+import { QuartileCandlestick } from "@hpcc-js/chart";
+import { Widget } from "@hpcc-js/common";
+import { Result, Workunit, WUInfo } from "@hpcc-js/comms";
+import { Grid } from "@hpcc-js/layout";
+import { Html } from "@hpcc-js/other";
+import { BreakdownTable, StyledTable } from "@hpcc-js/html";
+
+const knownProfileField = (sch: WUInfo.ECLSchemaItem): boolean => ["attribute", "given_attribute_type", "best_attribute_type", "rec_count", "fill_count", "fill_rate", "cardinality", "cardinality_breakdown", "modes", "min_length", "max_length", "ave_length", "popular_patterns", "rare_patterns", "is_numeric", "numeric_min", "numeric_max", "numeric_mean", "numeric_std_dev", "numeric_lower_quartile", "numeric_median", "numeric_upper_quartile", "numeric_correlations"].indexOf(sch.ColumnName) > 0;
+const countProfileFields = (r: Result): number => r.ECLSchemas.ECLSchemaItem.filter(knownProfileField).length;
+const isProfileResult = (r: Result, threshold = 4): boolean => countProfileFields(r) >= threshold;
+
+const config = {
+    rowHeight: 180,
+    primaryFontSize: 20,
+    secondaryFontSize: 14,
+    primaryColor: "#494949",
+    secondaryColor: "#DDD",
+    offwhiteColor: "#FBFBFB",
+    blueColor: "#1A99D5"
+};
+
+export class Test extends Grid {
+
+    _wu: Workunit;
+    _data: any[];
+
+    constructor(private _espUrl: string) {
+        super();
+        this
+            .gutter(0)
+            .surfaceShadow(false)
+            .surfacePadding("0")
+            .surfaceBorderWidth(0)
+            ;
+
+        const urlParts = this._espUrl.split("/WsWorkunits/res/");
+        const baseUrl = urlParts[0];
+        const urlParts2 = urlParts[1].split("/report/res/index.html");
+        const wuid = urlParts2[0];
+        this._wu = Workunit.attach({ baseUrl }, wuid);
+    }
+
+    enter(domNode, element) {
+        super.enter(domNode, element);
+        domNode.style.height = (this._data.length * config.rowHeight) + "px";
+
+        this._data.forEach((row, i) => {
```

Comment:
It's a game-changer to be able to do this type of thing within `enter`. If certain widget classes (like this one) could be designated as having "static data" then it greatly simplifies the concept of creating widget mashups.

Reply:
Whoever writes the Widget gets to dictate the shape of the data it expects (but obviously within the viz framework we want to keep it as consistent as possible)...